client/src/app/modules/events/event-edit/event-edit.component.ts

```diff
@@ -1,66 +1,169 @@
+  public cancel(): void {
     this.router.navigate(['/']);
   }
 
+  public getEvent(): Observable<IEventSchema> {
+    return this.eventService.getEvent(this.route.snapshot.params['id']);
+  }
+
+  private setIsCreateState(): void {
+    if (this.route.snapshot.params['id'] === 'new') {
+      this.isCreate = true;
+    } else {
+      this.isCreate = false;
+    }
+  }
+
+  /**
+   * Binding Event object to form model
+   */
+  private updateEventSchema(): void {
+    const toBoolean = (value: String): boolean =>  !value || value === 'false' ? false : true;
+    this.iEvent.name = this.eventForm.controls.name.value;
+    // FIXME: probably needed only for new event attributes which are not in formerly created events, can be removed later
+    if (!this.iEvent.location) {
+      this.iEvent.location = new IEventLocation();
+    }
+    this.iEvent.location.address = this.eventForm.controls.location.value;
+    this.iEvent.startDate = this.fromBrowserDate(this.eventForm.controls.dateStart.value);
+    this.iEvent.endDate = this.fromBrowserDate(this.eventForm.controls.dateEnd.value);
+    this.iEvent.description = this.eventForm.controls.description.value;
+    this.iEvent.limit = this.eventForm.controls.participantsLimit.value;
+    this.iEvent.transportBus = toBoolean(this.eventForm.controls.transport.value.bus);
+    this.iEvent.transportCar = toBoolean(this.eventForm.controls.transport.value.car);
+    this.iEvent.accommodation = toBoolean(this.eventForm.controls.accommodation.value);
+  }
+
+  /**
+   * Structure definition needed for template rendering
+   */
+  private getEventForm(): FormGroup {
+    return new FormGroup({
+      'name': new FormControl(null),
+      'location': new FormControl(null),
+      'dateStart': new FormControl(null),
+      'dateEnd': new FormControl(null),
+      'description': new FormControl(null),
+      'participantsLimit': new FormControl(null),
+      'transport': new FormGroup({
+        'bus': new FormControl(null),
+        'car': new FormControl(null)
+      }),
+      'accommodation': new FormControl(null)
+    });
+  }
+
+  private setEventFormValues(iEvent: IEventSchema): void {
+    const todayString = this.toBrowserDate(new Date());
+    const valueOrDefault = (value: any, defaultValue?: any): any => value ? value : defaultValue ? defaultValue : '';
+    this.eventForm.setValue({
+      'name': valueOrDefault(iEvent.name),
+      'location': iEvent.location ? valueOrDefault(iEvent.location.address) : '',
+      'dateStart': valueOrDefault(this.toBrowserDate(iEvent.startDate), todayString),
+      'dateEnd': valueOrDefault(this.toBrowserDate(iEvent.endDate), todayString),
+      'description': valueOrDefault(iEvent.description),
+      'participantsLimit': valueOrDefault(iEvent.limit),
+      transport: {
+        'car': valueOrDefault(iEvent.transportBus),
+        'bus': valueOrDefault(iEvent.transportCar)
+      },
+      'accommodation': valueOrDefault(iEvent.accommodation.toString())
```

Comment:
this will lead to a error, when the iEvent.accomodation is undefined (happened to me, when I reloaded the page of a old created event

Reply:
I already fixed this but didn't push it soon enough for you to see 😄